Source/JavaScriptCore/API/glib/JSCDefines.h.in

```diff
@@ -47,4 +47,30 @@
 #define JSC_DEPRECATED_FOR(f) JSC_API G_DEPRECATED_FOR(f)
 #endif /* !__GI_SCANNER__ */
 
+#if ENABLE(2022_GLIB_API)
+#define JSC_DECLARE_FINAL_TYPE(ModuleObjName, module_obj_name, MODULE, OBJ_NAME, ParentName) \
+    JSC_API G_DECLARE_FINAL_TYPE(ModuleObjName, module_obj_name, MODULE, OBJ_NAME, ParentName) \
+                                                                           \
+    typedef struct _ ## ModuleObjName ## Private ModuleObjName ## Private; \
+                                                                           \
+    struct _ ## ModuleObjName {                                            \
+        ParentName parent;                                                 \
+        /*< private >*/                                                    \
+        ModuleObjName ## Private *priv;                                    \
+    };
```

Comment:
Can't this be moved to the cpp?

Reply:
We need the declarations generated by the `G_DECLARE_FINAL_TYPE` macro to be part of the public headers, and I tried to move the instance struct to the implementation side but I hit some roadblock so I thought of doing it in a follow-up patch. I can give it a try again today after, maybe after sleeping the night I will manage to get it done 😉️

Reply:
Should definitely try to move the instance structs of final types to the cpp.

Reply:
(only in the new APi version, of course)

Reply:
Yeah so this definitely does not belong here because it prevents us from ever getting rid of `priv` during the lifetime of the new API. There are two better ways to do it:

 * Open code it in each .cpp file inside `#if ENABLE(2022_GLIB_API)` guards.
 * Have `WEBKIT_DEFINE_FINAL_TYPE` do it. I think this is probably nicest.

Reply:
My plan was to try piggybacking this into `WEBKIT_DEFINE_FINAL_TYPE` 🛠️&mdash;please bear with me because every set of changes involves making at least three different builds (WPE, GTK3, GTK4) to make sure things go in the right direction.